torch/nn/quantized/modules/linear.py

```diff
@@ -3,6 +3,67 @@
 from ...modules.module import Module
 from ...._jit_internal import weak_module
 
+@weak_module
+class Quantize(Module):
+    r"""Quantizes an incoming tensor
+    Args:
+     `output_scale`: scale of the output Quantized Tensor
+     `output_zero_point`: zero_point of output Quantized Tensor
+     `output_dtype`: data type of output Quantized Tensor
+
+    Attributes:
+      `output_scale`, `output_zero_point`, `output_dtype`
+
+    Examples::
+        >>> t = torch.tensor([[1., -1.], [1., -1.]])
+        >>> scale, zero_point, dtype = 1.0, 2, torch.qint8
+        >>> qm = Quantize(scale, zero_point, dtype)
+        >>> qt = qm(t)
+        >>> print(qt)
+        >>> tensor([[ 1., -1.],
+>         [ 1., -1.]], size=(2, 2), dtype=torch.qint8, scale=1.0, zero_point=2)
+    """
+
+    def __init__(self, output_scale, output_zero_point, output_dtype):
+        super(Quantize, self).__init__()
+        self.register_buffer('output_scale', torch.tensor([output_scale]))
+        self.register_buffer('output_zero_point', torch.tensor([output_zero_point], dtype=torch.long))
+        self.output_dtype = output_dtype
+
+    def forward(self, X):
+        Xq = torch.quantize_linear(X, self.output_scale.item(), self.output_zero_point.item(), self.output_dtype)
+        return Xq
+
+    @staticmethod
+    def from_float(mod):
+        return Quantize(mod.qparams[0].item(), mod.qparams[1].item(), torch.quint8)
+
+@weak_module
+class DeQuantize(Module):
```

Comment:
technically, it doesn't even need to be a module as it doesn't have any state

Reply:
yeah, but I think we need be consistent?